### PR TITLE
Update Arabic translation

### PR DESCRIPTION
**Pull request by SafaAlfulaij**
#2405

### CODE REVIEW

File: src/translations/sqlb_ar_SA.ts [hunk 1]
```diff
@@ -1019,7 +1100,7 @@ Errors are indicated with a red squiggle underline.</source>
         <location filename="../EditDialog.ui" line="290"/>
         <location filename="../EditDialog.ui" line="308"/>
         <source>Ctrl+P</source>
-        <translation></translation>
+        <translation>Ctrl+P</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:1103]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 2]
```diff
@@ -1039,7 +1120,7 @@ Errors are indicated with a red squiggle underline.</source>
     <message>
         <location filename="../EditDialog.ui" line="326"/>
         <source>Ctrl+Shift+C</source>
-        <translation></translation>
+        <translation>Ctrl+Shift+C</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:1123]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 3]
```diff
@@ -1771,18 +1852,18 @@ All data currently stored in this field will be lost.</source>
         <location filename="../ExtendedScintilla.cpp" line="62"/>
         <location filename="../ExtendedScintilla.cpp" line="283"/>
         <source>Ctrl+H</source>
-        <translation></translation>
+        <translation>Ctrl+H</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:1855]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 4]
```diff
@@ -1771,18 +1852,18 @@ All data currently stored in this field will be lost.</source>
...
         <source>Ctrl+F</source>
-        <translation></translation>
+        <translation>Ctrl+F</translation>
     </message>
     <message>
         <location filename="../ExtendedScintilla.cpp" line="77"/>
         <location filename="../ExtendedScintilla.cpp" line="287"/>
         <source>Ctrl+P</source>
-        <translation></translation>
+        <translation>Ctrl+P</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:1866]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 5]
```diff
@@ -2508,7 +2589,7 @@ You can drag SQL statements from an object row and drop them into other applicat
     <message>
         <location filename="../MainWindow.ui" line="2080"/>
         <source>Ctrl+/</source>
-        <translation></translation>
+        <translation>Ctrl+/</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:2592]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 6]
```diff
@@ -2600,7 +2681,7 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
     <message>
         <location filename="../MainWindow.ui" line="1554"/>
         <source>Shift+F5</source>
-        <translation></translation>
+        <translation>Shift+F5</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:2684]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 7]
```diff
@@ -2716,7 +2797,7 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
     <message>
         <location filename="../MainWindow.ui" line="1133"/>
         <source>Ctrl+N</source>
-        <translation></translation>
+        <translation>Ctrl+N</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:2800]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 8]
```diff
@@ -2743,7 +2824,7 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
     <message>
         <location filename="../MainWindow.ui" line="1157"/>
         <source>Ctrl+O</source>
-        <translation></translation>
+        <translation>Ctrl+O</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:2827]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 9]
```diff
@@ -2756,10 +2837,10 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
         <translation>يُغلق هذا الزر الاتصال بملف قاعدة البيانات المفتوح حاليًا</translation>
     </message>
     <message>
-        <location filename="../MainWindow.cpp" line="218"/>
-        <location filename="../MainWindow.cpp" line="3347"/>
+        <location filename="../MainWindow.cpp" line="227"/>
+        <location filename="../MainWindow.cpp" line="3429"/>
         <source>Ctrl+W</source>
-        <translation></translation>
+        <translation>Ctrl+W</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:2843]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 10]
```diff
@@ -2796,7 +2877,7 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
     <message>
         <location filename="../MainWindow.ui" line="1235"/>
         <source>Ctrl+S</source>
-        <translation></translation>
+        <translation>Ctrl+S</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:2880]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 11]
```diff
@@ -2822,7 +2903,7 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
     <message>
         <location filename="../MainWindow.ui" line="1266"/>
         <source>Ctrl+Q</source>
-        <translation></translation>
+        <translation>Ctrl+Q</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:2906]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 12]
```diff
@@ -2954,17 +3035,27 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
     <message>
         <location filename="../MainWindow.ui" line="1479"/>
         <source>Ctrl+T</source>
-        <translation></translation>
+        <translation>Ctrl+T</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:3038]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 13]
```diff
@@ -2979,12 +3070,12 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
     <message>
         <location filename="../MainWindow.ui" line="1184"/>
         <source>Ctrl+F4</source>
-        <translation></translation>
+        <translation>Ctrl+F4</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:3073]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 14]
```diff
@@ -2979,12 +3070,12 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
...
     </message>
     <message>
         <location filename="../MainWindow.ui" line="1448"/>
         <source>Shift+F1</source>
-        <translation></translation>
+        <translation>Shift+F1</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:3078]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 15]
```diff
@@ -3088,7 +3179,7 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
     <message>
         <location filename="../MainWindow.ui" line="1798"/>
         <source>Ctrl+Shift+O</source>
-        <translation></translation>
+        <translation>Ctrl+Shift+O</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:3182]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 16]
```diff
@@ -3124,7 +3215,7 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
     <message>
         <location filename="../MainWindow.ui" line="1840"/>
         <source>Ctrl+F</source>
-        <translation></translation>
+        <translation>Ctrl+F</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:3218]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 17]
```diff
@@ -3145,7 +3236,7 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
     <message>
         <location filename="../MainWindow.ui" line="1867"/>
         <source>Ctrl+H</source>
-        <translation></translation>
+        <translation>Ctrl+H</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:3239]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 18]
```diff
@@ -3302,7 +3393,7 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
     <message>
         <location filename="../MainWindow.ui" line="2137"/>
         <source>Ctrl+Shift+S</source>
-        <translation></translation>
+        <translation>Ctrl+Shift+S</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:3396]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 19]
```diff
@@ -3365,9 +3454,9 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
         <translation>احفظ ملف SQL</translation>
     </message>
     <message>
-        <location filename="../MainWindow.cpp" line="282"/>
+        <location filename="../MainWindow.cpp" line="291"/>
         <source>Ctrl+E</source>
-        <translation></translation>
+        <translation>Ctrl+E</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:3459]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 20]
```diff
@@ -3425,69 +3514,69 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
     <message>
         <location filename="../MainWindow.ui" line="1497"/>
         <source>Ctrl+Return</source>
-        <translation></translation>
+        <translation>Ctrl+Enter</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:3517]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 21]
```diff
@@ -3425,69 +3514,69 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
...
     </message>
     <message>
-        <location filename="../MainWindow.cpp" line="267"/>
+        <location filename="../MainWindow.cpp" line="276"/>
         <source>Ctrl+L</source>
-        <translation></translation>
+        <translation>Ctrl+L</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:3522]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 22]
```diff
@@ -3425,69 +3514,69 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
...
     </message>
     <message>
         <location filename="../MainWindow.ui" line="2029"/>
         <location filename="../MainWindow.ui" line="2056"/>
         <source>Ctrl+P</source>
-        <translation></translation>
+        <translation>Ctrl+P</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:3528]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 23]
```diff
@@ -3425,69 +3514,69 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
...
     </message>
     <message>
-        <location filename="../MainWindow.cpp" line="272"/>
+        <location filename="../MainWindow.cpp" line="281"/>
         <source>Ctrl+D</source>
-        <translation></translation>
+        <translation>Ctrl+D</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:3533]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 24]
```diff
@@ -3425,69 +3514,69 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
...
     </message>
     <message>
-        <location filename="../MainWindow.cpp" line="277"/>
+        <location filename="../MainWindow.cpp" line="286"/>
         <source>Ctrl+I</source>
-        <translation></translation>
+        <translation>Ctrl+I</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:3538]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 25]
```diff
@@ -3496,310 +3585,359 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
...
+        <location filename="../MainWindow.cpp" line="330"/>
         <source>Reset Window Layout</source>
         <translation>صفّر تخطيط النافذة</translation>
     </message>
     <message>
-        <location filename="../MainWindow.cpp" line="318"/>
+        <location filename="../MainWindow.cpp" line="331"/>
         <source>Alt+0</source>
-        <translation></translation>
+        <translation>Alt+0</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:3730]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 26]
```diff
@@ -3496,310 +3585,359 @@ You can drag SQL statements from the Schema column and drop them into the SQL ed
...
+    <message>
+        <location filename="../MainWindow.cpp" line="336"/>
+        <source>Simplify Window Layout</source>
+        <translation>بسّط تخطيط النافذة</translation>
     </message>
     <message>
-        <location filename="../MainWindow.cpp" line="355"/>
+        <location filename="../MainWindow.cpp" line="337"/>
+        <source>Shift+Alt+0</source>
+        <translation>Shift+Alt+0</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:3740]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 27]
```diff
@@ -3991,7 +4134,7 @@ Create a backup!</source>
     <message>
         <location filename="../AddRecordDialog.cpp" line="44"/>
         <source>Alt+Del</source>
-        <translation></translation>
+        <translation>Alt+Del</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:4137]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 28]
```diff
@@ -5460,7 +5963,7 @@ p, li { white-space: pre-wrap; }
     <message>
         <location filename="../SqlExecutionArea.ui" line="91"/>
         <source>Shift+F3</source>
-        <translation></translation>
+        <translation>Shift+F3</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:5966]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 29]
```diff
@@ -5570,7 +6073,7 @@ p, li { white-space: pre-wrap; }
     <message>
         <location filename="../sqltextedit.cpp" line="41"/>
         <source>Ctrl+/</source>
-        <translation></translation>
+        <translation>Ctrl+/</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:6076]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 30]
```diff
@@ -5999,7 +6502,7 @@ Hold %3Shift and click to jump there</source>
     <message>
         <location filename="../TableBrowser.ui" line="276"/>
         <source>Shift+F3</source>
-        <translation></translation>
+        <translation>Shift+F3</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:6505]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 31]
```diff
@@ -6407,7 +6910,7 @@ Hold %3Shift and click to jump there</source>
     <message>
         <location filename="../TableBrowser.ui" line="803"/>
         <source>Ctrl+P</source>
-        <translation></translation>
+        <translation>Ctrl+P</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:6913]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 32]
```diff
@@ -6441,7 +6944,7 @@ Hold %3Shift and click to jump there</source>
     </message>
     <message>
         <source>Ctrl+F</source>
-        <translation></translation>
+        <translation type="vanished">Ctrl+F</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:6947]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

**but this one is vanished entry so you can ignore this one.**

File: src/translations/sqlb_ar_SA.ts [hunk 33]
```diff
@@ -6452,7 +6955,7 @@ Hold %3Shift and click to jump there</source>
     <message>
         <location filename="../TableBrowser.ui" line="860"/>
         <source>Ctrl+B</source>
-        <translation></translation>
+        <translation>Ctrl+B</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:6958]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 34]
```diff
@@ -6469,7 +6972,7 @@ Hold %3Shift and click to jump there</source>
     <message>
         <location filename="../TableBrowser.ui" line="893"/>
         <source>Ctrl+U</source>
-        <translation></translation>
+        <translation>Ctrl+U</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:6975]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 35]
```diff
@@ -6558,7 +7061,7 @@ Hold %3Shift and click to jump there</source>
     <message>
         <location filename="../TableBrowser.ui" line="1039"/>
         <source>Ctrl+Space</source>
-        <translation></translation>
+        <translation>Ctrl+Space</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:7064]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 36]
```diff
@@ -6567,20 +7070,24 @@ Hold %3Shift and click to jump there</source>
     </message>
     <message>
         <source>Ctrl+H</source>
-        <translation></translation>
+        <translation type="vanished">Ctrl+H</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:7073]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

**but this one is vanished entry so you can ignore this one.**

File: src/translations/sqlb_ar_SA.ts [hunk 37]
```diff
@@ -6567,20 +7070,24 @@ Hold %3Shift and click to jump there</source>
...
+    <message>
+        <location filename="../TableBrowser.cpp" line="42"/>
+        <source>Filter in any column</source>
+        <translation>رشّح أيّ عمود</translation>
     </message>
     <message>
         <location filename="../TableBrowser.cpp" line="102"/>
         <source>Ctrl+R</source>
-        <translation></translation>
+        <translation>Ctrl+R</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:7087]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

File: src/translations/sqlb_ar_SA.ts [hunk 38]
```diff
@@ -6592,113 +7099,113 @@ Hold %3Shift and click to jump there</source>
...
+        <location filename="../TableBrowser.cpp" line="1177"/>
         <source>Duplicate record</source>
         <translation>كرّر السجلّ</translation>
     </message>
     <message>
-        <location filename="../TableBrowser.cpp" line="1172"/>
+        <location filename="../TableBrowser.cpp" line="1181"/>
         <source>Ctrl+&quot;</source>
-        <translation></translation>
+        <translation>Ctrl+&quot;</translation>
```
Comment (lucydodo) [inline src/translations/sqlb_ar_SA.ts:7161]: Shortcuts should not be translated.
See the [wiki](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations#dont-translate-key-names-in-shortcuts) for further information.

**Review by lucydodo — CHANGES_REQUESTED** [review]
Thank you for contributing! :smile: 
I found some items that needed simple corrections.
I could only edit those parts and merge them, but I left it as a review only as you may have disagreements. Check them please.

**Comment by SafaAlfulaij** [comment]
Hi @lucydodo 
I can't see how "just leave the translation empty and finished" will make Qt use the original shortcut.
I tried this locally and the shortcut won't work any more, and it's text disappeared.

image

Even if Qt for some reason or another used the orignal string (if the translation is empty), this _must_ be reported as a bug.
If I say that this string is empty, then it must be empty. Translations overrides original.
Shortcut or not, that doesn't matter.

**Comment by lucydodo** [comment]
```
<source>Ctrl+N</source>
<translation></translation>
```

Interesting.. 🤔 Did you do the like as above when testing locally?
I just installed the Arabic version of Windows 10 on VM and installed the latest nightly build to test it out. (All shortcuts are not translated in current master branch) However as in the screenshot below, shortcuts are displayed, and they work fine.

شش

**Comment by justinclift** [comment]
@SafaAlfulaij 

> I can't see how "just leave the translation empty and finished" will make Qt use the original shortcut.

Yeah, I wasn't sure about that either but (apparently) it's a feature of Qt (?).  eg when the translated string is empty, it automatically uses the original text instead

@mgrojo That's correct yeah?

**Comment by mgrojo** [comment]
> @SafaAlfulaij
> 
> > I can't see how "just leave the translation empty and finished" will make Qt use the original shortcut.
> 
> Yeah, I wasn't sure about that either but (apparently) it's a feature of Qt (?). eg when the translated string is empty, it automatically uses the original text instead
> 
> @mgrojo That's correct yeah?

Yes, this is what I know due to experience, but I haven't found any reference in the documentation. In the [Qt Linguist Manual](https://doc.qt.io/archives/qt-4.8/linguist-translators.html) there are references to "accept the empty translation", but without clarifying what is then shown in the translation. Having said that, I don't have an explanation for @SafaAlfulaij's screenshot. For me, that Qt version is buggy, since it goes against all what we have seen till now. This is my case with an empty translation for "Ctrl+N":
image

**Comment by SafaAlfulaij** [comment]
Filed a bug upstream: [#86774](https://bugreports.qt.io/browse/QTBUG-86774)

**Comment by lucydodo** [comment]
@SafaAlfulaij But what about the nightly build? Even there, the shortcut doesn't appear and doesn't work?

**Comment by SafaAlfulaij** [comment]
It seems that the translation file was changed from the last time I updated it...

This is _not_ the correct behaviour. If the translation is empty, and finished, then it _must_ be empty in the application.

And as for the nightly build, yes the shortcuts are there and work, but if you take a closer look to the exe:
image

**("ÿÿÿÿ" = "FF FF FF FF")**

If you believe as I believe, that "-1" is an invalid string length, then let's please fix the wrong "standard" established within this project.
If not, I'm not sure what to say...

**Comment by mgrojo** [comment]
If this is a Qt bug, I don't know what they're doing about translation of keyboard shortcuts. Our recommendation to not translate them comes from various issues that we have received (see, for example, #2178). But I'm not sure if we have received some issue when the translation string is equal to the original. Maybe not and, in that case, we could merge this pull request without problem. But what about the other languages, are people building with Qt 5.15 going to have problems when the translation of shortcuts is empty? What Qt version are we using for the official builds?

**Comment by justinclift** [comment]
> What Qt version are we using for the official builds?

At the moment, it's Qt 5.12.8.  We could update that to Qt 5.12.9, but I kind of figured... "why take the chance of possibly breaking something for our supposed-to-be-stable point release?". :wink:

**Comment by mgrojo** [comment]
> > What Qt version are we using for the official builds?
> 
> At the moment, it's Qt 5.12.8. We could update that to Qt 5.12.9, but I kind of figured... "why take the chance of possibly breaking something for our supposed-to-be-stable point release?". wink

Perfect. I only wanted to confirm that it isn't 5.15, which is apparently the one causes problems.

**Comment by SafaAlfulaij** [comment]
First, sorry if I was rude in any way.
I'll explain in details:
- Empty approved translations should show empty in the app (my screenshot above).
- lrelease has a bug that if the translation is empty and approved, it will give it a length of -1 in the released QM file, which will (in some way or another) make `QTranslator` skips the translation as if it's not there and use the original.
- DB4S is recommending translators to _use_ this bug (without knowing it's a bug in the first place), instead of letting them know about shortcuts and the proper way to translate them.

Regarding #2178, I'm not sure what the problem is, but perhaps it's because of keyboard layouts or something else. Did you try Ctrl+Enter and Ctrl+Return? These two are different keys (return is the normal, Enter is the keypad). Perhaps this was the problem?
Normally devs assign both as sequences for the same action (in code).

**Comment by justinclift** [comment]
> lrelease has a bug that if the translation is empty and approved, it will give it a length of -1 in the released QM file, which will (in some way or another) make QTranslator skips the translation as if it's not there and use the original.

Do we know for sure that it **is** a bug, and not something that's understood to work this way + just badly documented?

**Comment by mgrojo** [comment]
In fact, we never said that shortcuts must not be translated, but that key names should not be translated, because this is known not to work and is more or less [documented](https://doc.qt.io/qt-5/linguist-translators.html#ctrl-key-accelerators).

> Warning: Do not translate the "Alt", "Ctrl" or "Shift" parts of the accelerators. Qt relies on these strings being there. For supported languages, Qt automatically translates these strings.

(They should say that you are not supposed to translate key names in general, like Space or Return).

But I think we haven't actually have problems when the translation is equal to the original. Emptying the translation was just our quick way to avoid the problem in all languages. On the other hand, yes, I'm sure that leaving the translation empty and finished is, in general, not only for shortcuts, saying to Qt that there's no translation and the original text is to be kept. They have designed it that way, otherwise it's a so long-standing bug, that it has become a feature.

I think the pull request can be merged as is.

**Review by lucydodo — APPROVED** [review]

**Comment by lucydodo** [comment]
Based on the commit of @SafaAlfulaij, I tested it in Windows and macOS, and confirmed that it works.
>This is because on macOS the `Ctrl` key sequence is replaced by the `Command` key sequence.

In fact, I think it's okay to merge in this case, but on the other hand, we have been doing like this until this time, and the official Qt IRC channel also confirmed that if the blank translation was approved, replaced by the original text
(but the person who gave the answer was not the official Qt developer),

if assuming that this is not a bug, I requested changes because of concerns about bugs that may occur in the stable version.

First of all, I approved the review based on the comments arising from this PR.
but I delegated choice to someone else about merge.

**Comment by mgrojo** [comment]
It is now merged to master. I'll try to cherry-pick it to the v3.12.x, which is the branch of the release. I hope it can be done without probles.

**Comment by mgrojo** [comment]
@SafaAlfulaij I had to solve some trivial conflicts and it looks good from a technical point of view. When compiling, it says that there are two pending messages, but I cannot find them. You can review it in 1dfa70c209453ab3fdce7991377fdba94bafccff

**Comment by SafaAlfulaij** [comment]
@mgrojo Couldn't find it either...

**Pull request merged**